src/logging.c

```diff
@@ -86,6 +87,54 @@ rcutils_ret_t rcutils_logging_initialize(void)
   return rcutils_logging_initialize_with_allocator(rcutils_get_default_allocator());
 }
 
+enum rcutils_get_env_retval
+{
+  RCUTILS_GET_ENV_ERROR = -1,
+  RCUTILS_GET_ENV_ZERO = 0,
+  RCUTILS_GET_ENV_ONE = 1,
+  RCUTILS_GET_ENV_DEFAULT = 2,
+};
+
+// A utility function to get zero or one from an environment variable.  Returns
+// RCUTILS_GET_ENV_ERROR if we failed to get the environment variable or if it was
+// something we don't understand.  Return RCUTILS_GET_ENV_ZERO if the value in the
+// environment variable is "0", RCUTILS_GET_ENV_ONE if the value in the environment
+// variable is "1", or RCUTILS_GET_ENV_DEFAULT if the environment variables is empty.
+static enum rcutils_get_env_retval rcutils_get_env_var_zero_or_one(
+  const char * name, const char * zero_semantic,
+  const char * one_semantic)
+{
+  const char * env_var_value = NULL;
+  const char * ret_str = rcutils_get_env(name, &env_var_value);
+  if (NULL != ret_str) {
+    fprintf(
+      stderr, "Error getting environment variable "
+      "%s: %s\n", name, ret_str);
+    RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      "Error getting environment variable %s: %s", name,
+      ret_str);
+    return RCUTILS_GET_ENV_ERROR;
+  }
+
+  if (strcmp(env_var_value, "") == 0) {
+    return RCUTILS_GET_ENV_DEFAULT;
```

Comment:
This function doesn't have any concept of a default value. Therefore I would argue this return code should instead be "empty/not-set". It is up to the caller to determine the semantic of it.

Reply:
That's fine.  I'll change the name of the enum to `RCUTILS_GET_ENV_EMPTY`.